Lib/glyphsLib/builder/sources.py

```diff
@@ -53,6 +53,16 @@ def _to_designspace_source(self, master, is_regular):
             # FIXME: (jany) have this function not write the dot
             build_ufo_path('.', source.familyName, source.styleName))
 
+        # Make sure UFO filenames are unique, lest we overwrite masters that
+        # happen to have the same weight name.
+        if any(s != source and s.filename == source.filename
```

Comment:
what type are `s` and `source` here? do they implement `__eq__`? Maybe you meant `is not` for identity checking, rather than equality

Reply:
`s` is the same type as `source`, both are in the same list. I'm filtering out the element in question while looking at all other elements in the list. Will try `is not` :)